other/react-native/__tests__/render-tests.js

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable react/prop-types */
+// eslint-disable-next-line import/no-unassigned-import
+import {Text, TextInput, View} from 'react-native'
+import React from 'react'
+
+// Note: test renderer must be required after react-native.
+import TestRenderer from 'react-test-renderer'
+
+import Downshift from '../../../dist/downshift.native.cjs'
+
+test('renders with React Native components', () => {
+  const RootView = ({innerRef, ...rest}) => <View ref={innerRef} {...rest} />
+  const renderSpy = jest.fn(({getRootProps, getInputProps, getItemProps}) => (
+    <RootView {...getRootProps({refKey: 'innerRef'})}>
+      <TextInput {...getInputProps()} />
+      <View>
+        <Text {...getItemProps({item: 'foo', index: 0})}>foo</Text>
+        <Text {...getItemProps({item: 'bar', index: 1})}>bar</Text>
+      </View>
+    </RootView>
+  ))
+  const element = <Downshift render={renderSpy} />
+  const renderer = TestRenderer.create(element)
+  expect(renderSpy).toHaveBeenCalledWith(
+    expect.objectContaining({
+      isOpen: false,
+      highlightedIndex: null,
+      selectedItem: null,
+      inputValue: '',
+    }),
+  )
+  const tree = renderer.toJSON()
+  expect(tree).toMatchSnapshot()
+})
+
+test('can use children instead of render prop', () => {
+  const RootView = ({innerRef, ...rest}) => <View ref={innerRef} {...rest} />
+  const renderSpy = jest.fn(({getRootProps, getInputProps, getItemProps}) => (
+    <RootView {...getRootProps({refKey: 'innerRef'})}>
+      <TextInput {...getInputProps()} />
+      <View>
+        <Text {...getItemProps({item: 'foo', index: 0})}>foo</Text>
+        <Text {...getItemProps({item: 'bar', index: 1})}>bar</Text>
+      </View>
+    </RootView>
+  ))
+  const element = <Downshift>{renderSpy}</Downshift>
+  TestRenderer.create(element)
+  expect(renderSpy).toHaveBeenCalledTimes(1)
+})
+
+test('getInputProps composes onChange with onChangeText', () => {
```

Comment:
I think this is more incidental. I think most people will want to be able to pass `onChangeText` instead of `onChange` if that's what they're used to. Is that currently possible? Could we have our tests ensure that?

Reply:
Ah, this is actually a poor test, and one difference between React Native and ReactDOM that should be documented.

[`onInput` is not a valid prop for `TextInput`](https://facebook.github.io/react-native/docs/textinput.html), but in this case, Downshift is still composing the `onChange` prop with a new function for `onInput`, and then calling both. The `onInput` prop has no affect on the `TextInput` in this case.

To emulate this behavior on React Native, perhaps we would compose `onChange` with `onChangeText`, [where `onChangeText` is just a convenience wrapper for `onChange(event.nativeEvent.text)`](https://github.com/facebook/react-native/blob/6f007e8957c9bf5652b0184cba65f385050a8236/Libraries/Components/TextInput/TextInput.js#L925-L927). There's no other events that come down through the `onChange ` prop on `TextInput`, like there would be on `<input>` in ReactDOM; [`TextInput` has the prop `onSelectChange`](https://facebook.github.io/react-native/docs/textinput.html#onselectionchange) to handle things like that.

When I get to documenting this difference (as we talked about in #265), I'll be sure to call this out. We might even be able to optimize this a bit for React Native by forgoing any prop composition, since this case isn't quite valid for us anyway.

Right now, the behavior of the Downshift RN implementation is for `getInputProps` to accept `onChange` as the function and apply it to the `TextInput` as `onChangeText`. Since `onChange` and `onChangeText` have different arguments, perhaps it'd be better to make a special case for this in the RN implementation that allows for using either function with `getInputProps` and applying it accordingly.

What do you think?

Reply:
I want people to assume that the arguments to the prop getters to be the props they want applied. So if normally they would use `onChangeText={this.handleOnChangeText}` in their JSX, then they should use `onChangeText: this.handleOnChangeText` in the function call.

Reply:
What's the status here?

Reply:
Ah, sorry, the status of this is that I went on vacation for a bit 😬 

I think there's an API change that'll need to happen to address this (for React Native). When I originally implemented this React Native support, I didn't have a full grasp of the Downshift API, so I allowed for the same `onChange` prop to be supplied, but then used it as `onChangeText`.

I think the correct API way forward would be to support both `onChange` and `onChangeText`, but that may be out of the scope of this PR, which is to simply add a React Native test suite.

I'll make an issue to track this.

Reply:
I filed #361 to track this for now. I'll remove this test since it's not entirely valid.